code/core/src/manager-api/lib/merge.ts

```diff
@@ -1,23 +1,19 @@
-import mergeWith from 'lodash/mergeWith.js';
-import isEqual from 'lodash/isEqual.js';
+import { defu, defuArrayFn } from 'defu';
 
 import { logger } from '@storybook/core/client-logger';
 
-export default <TObj = any>(a: TObj, b: Partial<TObj>) =>
-  mergeWith({}, a, b, (objValue: TObj, srcValue: Partial<TObj>) => {
-    if (Array.isArray(srcValue) && Array.isArray(objValue)) {
-      srcValue.forEach((s) => {
-        const existing = objValue.find((o) => o === s || isEqual(o, s));
-        if (!existing) {
-          objValue.push(s);
-        }
-      });
-
-      return objValue;
-    }
-    if (Array.isArray(objValue)) {
-      logger.log(['the types mismatch, picking', objValue]);
-      return objValue;
+export default <TObj = any>(a: TObj, b: Partial<TObj> & any) => {
+  if (Array.isArray(a)) {
+    if (Array.isArray(b)) {
+      return defuArrayFn(a, b);
     }
-    return undefined;
-  });
+
+    logger.log(['the types mismatch, picking', a]);
+    return a;
+  }
+
+  // TODO: Ask for what is prefered:
+  // 1. type guard (safe at runtime, but potential performance slow-down)
+  // 2. or type assertion (unsafe at runtime)
```

Comment:
I need help with decision on this one.

I've had issues with types mismatch, when the records from the second argument had types which did not exists on the first argument _(source)_ in the `PreviewWeb.test.ts`.

Is it okay to change type of second argument from `Partial<TObj>` to `Partial<TObj> & any`?

Reply:
Sure. Just widen the types for now and we can take a proper look afterwards.